### PR TITLE
Allow disabling of the tracking via env var

### LINKED ISSUE #779: Make it possible to disable analytics via a env var.
This will help support the automated testing and ensure we don't skew analytics.

/cc @davehunt 

Relates to #793

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #779

### CODE REVIEW

File: tests/client/core/test_utils.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { camelCaseProps, getClientConfig } from 'core/utils';
+import { camelCaseProps, getClientConfig, getAsBool } from 'core/utils';
```
Comment (tofumatt) [inline tests/client/core/test_utils.js]: My OCD is noticing these aren't alphabetised 😉

**Comment by muffinresearch** [comment]
Hmm, I think we probably don't need to do this since we're going to need a uitest specific conf instead.

File: src/core/utils.js [hunk 2]
```diff
@@ -19,10 +20,23 @@ export function camelCaseProps(obj) {
...
-  for (const key of config.get('clientConfigKeys')) {
-    clientConfig[key] = config.get(key);
+  for (const key of _config.get('clientConfigKeys')) {
+    clientConfig[key] = _config.get(key);
   }
   return clientConfig;
 }
+
+export function getAsBool(key, { _config = config } = {}) {
+  const value = _config.get(key);
```
Comment (mstriemer) [inline src/core/utils.js]: I think I'd prefer this function to not have anything to do with `config`. So call it like:

```js
trackingEnabled: utils.asBool(config.get('trackingEnabled'))
```

It probably shouldn't have `get` in the name then. `asBool`, `convertBool`, `toBool` or something.

Comment (muffinresearch) [inline src/core/utils.js]: Ok, I suppose the benefit of that is it makes it clearer where the value is coming from.

File: tests/server/helpers.js [hunk 3]
```diff
@@ -13,11 +13,12 @@ export function checkSRI(res) {
   const $script = $('script[src]');
   assert.ok($script.length > 0, 'must be at least 1 script');
   $script.each((i, elem) => {
-    if (!$(elem).attr('src').includes('analytics.js')) {
-      assert.include($(elem).attr('integrity'),
-        'sha512', 'script should have integrity attr');
-      assert.equal($(elem).attr('crossorigin'),
-        'anonymous', 'script should have crossorigin attr');
+    if ($(elem).attr('src').includes('analytics.js')) {
```
Comment (mstriemer) [inline tests/server/helpers.js]: Might be nice to do `const $elem = $(elem);` at the top of this loop, looks like we're doing that work three times.

**Comment by mstriemer** [comment]
r+wc

**Pull request merged**